### PR TITLE
Hb 07

### PR DESCRIPTION
**Pull request by vremere-a**

### CODE REVIEW

File: src/main/java/com/dev/cinema/Main.java [hunk 1]
```diff
@@ -104,5 +105,11 @@ public static void main(String[] args) throws AuthenticationException {
         shoppingCartService.addSession(spiderManSession,user2);
         System.out.println(shoppingCartService.getByUser(user2));
         shoppingCartService.clear(shoppingCartService.getByUser(user2));
+
+        OrderService orderService =
+                (OrderService) injector.getInstance(OrderService.class);
+        orderService.completeOrder(shoppingCartService.getByUser(user3).getTickets(), user3);
+        System.out.println(orderService.getOrderHistory(user3));
+
     }
```
Comment (dariiabutorina) [inline src/main/java/com/dev/cinema/Main.java]:
```suggestion
        System.out.println(orderService.getOrderHistory(user3));
    }
```

**Review by dariiabutorina — APPROVED** [review]
Well done :)

File: src/main/java/com/dev/cinema/Main.java (hunk 1 above)
Comment (vremere-a) [reply src/main/java/com/dev/cinema/Main.java]: 👍

File: src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            transaction = session.beginTransaction();
+            session.save(order);
+            transaction.commit();
+            return order;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add order entity with id"
+                    + order.getId(), e);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java]:
```suggestion
            throw new DataProcessingException("Can't add order entity: "
                    + order, e);
```

if something goes wrong, will id be set?

File: src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<Order> orderCriteriaQuery =
+                    criteriaBuilder.createQuery(Order.class);
+            Root<Order> orderRoot =
+                    orderCriteriaQuery.from(Order.class);
+            orderRoot.fetch("tickets");
+            Predicate predicate =
+                    criteriaBuilder.equal(orderRoot.get("user"), user);
+            orderCriteriaQuery.select(orderRoot).where(predicate);
+            return session.createQuery(orderCriteriaQuery).getResultList();
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java]: let's return only distinct results here

File: src/main/java/com/dev/cinema/model/Order.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+@Table(name = "orders")
+public class Order {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @OneToMany
+    private List<Ticket> tickets;
+    private LocalDateTime orderDate;
+    @OneToOne
+    private User user;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/Order.java:27]: are you sure about this? how many orders can 1 user have?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java (hunk 2 above)
Comment (vremere-a) [reply src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java]: done

File: src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java (hunk 3 above)
Comment (vremere-a) [reply src/main/java/com/dev/cinema/dao/impl/OrderDaoImpl.java]: return session.createQuery(orderCriteriaQuery.distinct(true)).getResultList();
done

File: src/main/java/com/dev/cinema/model/Order.java (hunk 4 above)
Comment (vremere-a) [reply src/main/java/com/dev/cinema/model/Order.java:27]: sorry.  ok

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**